versionRuleset.xml

```diff
@@ -2,6 +2,11 @@
          xmlns="http://mojo.codehaus.org/versions-maven-plugin/rule/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://mojo.codehaus.org/versions-maven-plugin/rule/2.0.0 https://www.mojohaus.org/versions-maven-plugin/xsd/rule-2.0.0.xsd">
 <rules>
+    <rule groupId="*" artifactId="*s" comparisonMethod="maven">
+       <ignoreVersions>
+          <ignoreVersion type="regex">^(?!.[\s\S]*-M\d*).[\s\S]*.*[a-zA-Z]\d.*$</ignoreVersion>
```

Comment:
This should be excluding all rc items as well.  At least I have not been seeing any in my updates recently.

Reply:
If that regex is for what versions should be ignored--which it seems from the name--then it does seem to me that RC versions will be ignored. It's more likely that it's rare that dev teams just upload Release Candidates to Maven Central and so amongst all our direct and transitive dependencies, we just aren't seeing any. But consider our own 2.2.1.0-RC1 release. It doesn't seem like this regex would match that and--if I'm understanding things right--that it therefore wouldn't be ignored. Of course, it's just as likely that I'm misinterpreting the whole <ignoreVersion> attribute.

Reply:
I think the reverse look-ahead is confusing, but it does match 2.2.1.0-RC1
I just tested it out at [regex101](https://regex101.com/)

Reply:
Ah; yeah. I didn't notice the look-ahead '(?' in  the regex. Read right past it. :-( I generally try to avoid that because it is subtle and often prone to ReDoS attacks. ReDoS isn't a concern here though.